src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelConnectionOfT.cs

```diff
@@ -37,13 +38,23 @@ void IThreadPoolWorkItem.Execute()
     internal async Task ExecuteAsync()
     {
         var connectionContext = _transportConnection;
+        var metricsConnectionDurationEnabled = _serviceContext.Metrics.IsConnectionDurationEnabled();
```

Comment:
Is this new code going to affect AOT/trimming.  If so, this doesn't seem like an analyzable way to disable it.

Reply:
Metrics won't be trimmable. But if the size increase is significant then we'll reevaluate.